### ISSUE TITLE
SetData and scripting interfaces

### ISSUE THREAD

**Issue by openbabel-bot** [openbabel/openbabel#782]
We need to add a CopyData() method in base.cpp:

There's a fundamental problem with using SWIG to wrap methods that
take and keep instances of objects created by the user. The only
example I know of in OpenBabel is SetData.

In Python, this is used as follows:
pairdata = ob.OBPairData()
pairdata.SetAttribute("MolWeight")
pairdata.SetValue("49.0")
pairdata.thisown = 0 # So that SWIG Proxy will not delete pairdata
mol.SetData(pairdata)

Without line 4, a segfault occurs, as SWIG deletes pairdata once it
goes out of local scope, despite the fact that 'mol' has a reference
to it.

Actually, line 4 is a bit of a hack. In Java, it's just not possible
to do this, as 'thisown' is a protected attribute and not available
through the API (after reading the SWIG docs, it seems that a massive
hack would be required to work around this). Similarly, this will be a
problem with other scripting languages.

In short, is there an alternative way around this on the C++ side? For
example, if SetData cloned the pairdata. As this is likely to be a
general problem, are there any other known instances of this?

Reported by: @ghutchis
